### ISSUE TITLE
Proposal: SA1027 should still reports tabs after non-whitespace with useTabs

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#2035]
~~**Blocked on the acceptance of #2039, which adds the option described here.**~~

#2039 Adds a new `useTabs` option which defaults to `false`. When this value is set to true, SA1027 (TabsMustNotBeUsed) is automatically disabled.

I propose updating the implementation when `useTabs` is true to only disable analysis of tabs used for indentation. In other words, tabs placed after the first non-whitespace character on a line would still be reported.

This is related to the OneTabIndent option described in #730, but not quite the same. This proposal only covers a specific case where the use of tab characters *always* produces strange behavior when the tab width changes.

**Comment by vweijsters** [comment]
:+1:

**Comment by sharwell** [comment]
This proposal is no longer blocked.

**Comment by AArnott** [comment]
Can you give an example of bad use of tabs after non-whitespace characters? Are we thinking something like:

```csharp
int foo;<TAB><TAB>// my first
int barbaz;<T><TB>// my second
```

Where we want folks to use spaces instead of tabs?
While that *may* be a benefit, it's perhaps a tall order on the programmer to have their IDE tab setting be to actually insert tab characters (for the typical indentation) but avoid use of the TAB key for other indentation use. However, as you say, if the TAB size were to be changed, in my above example it could potentially make the indentation of the comments look very weird, and the space key is definitely one that programmers are comfortable with :smile:. So I guess I support.

**Comment by sharwell** [comment]
@AArnott The IDE does not have an exact setting matching this behavior, but I believe it's acceptable for two reasons:

1. The use of tab within a line is *relatively* infrequent
2. The analysis is fast, and fixing all violations at once is fast. The fixer also ensures the conversion from tabs to spaces preserves the relative positioning of items within a line.

**Comment by AArnott** [comment]
Sounds good. :+1:

**Comment by Noryoko** [comment]
:+1:
